src/libAtomVM/nifs.c

```diff
@@ -5842,6 +5868,118 @@ static term nif_erlang_lists_subtract(Context *ctx, int argc, term argv[])
     return result;
 }
 
+
+static term nif_lists_member(Context *ctx, int argc, term argv[])
+{
+    UNUSED(argc)
+    term elem = argv[0];
+    term list = argv[1];
+    VALIDATE_VALUE(list, term_is_list);
+
+    int proper;
+    term_list_length(list, &proper);
+    if (UNLIKELY(!proper)) {
+        RAISE_ERROR(BADARG_ATOM);
+    }
+
+    while (!term_is_nil(list)) {
+        term head = term_get_list_head(list);
+
+        TermCompareResult cmp_result = term_compare(head, elem, TermCompareExact, ctx->global);
+
+        if (cmp_result == TermEquals) {
+            return TRUE_ATOM;
+        }
+
+        if (UNLIKELY(cmp_result == TermCompareMemoryAllocFail)) {
+            RAISE_ERROR(OUT_OF_MEMORY_ATOM);
+        }
+
+        list = term_get_list_tail(list);
+    }
+
+    return FALSE_ATOM;
+}
+
+static term nif_lists_keymember(Context *ctx, int argc, term argv[])
+{
+    term result = nif_lists_keyfind(ctx, argc, argv);
+    return result == FALSE_ATOM ? FALSE_ATOM : TRUE_ATOM;
+}
+
+static term nif_lists_keyfind(Context *ctx, int argc, term argv[])
+{
+    UNUSED(argc)
+    term key = argv[0];
+    term n = argv[1];
+    term tuple_list = argv[2];
+
+    if (!term_is_integer(n) || !term_is_list(tuple_list)) {
+        RAISE_ERROR(BADARG_ATOM);
+    }
+
+    int n_pos = term_to_int(n);
+
+    if (n_pos <= 0) {
+        RAISE_ERROR(BADARG_ATOM);
+    }
+
+    int proper;
+    term_list_length(tuple_list, &proper);
+    if (UNLIKELY(!proper)) {
+        RAISE_ERROR(BADARG_ATOM);
+    }
+
+    while (!term_is_nil(tuple_list)) {
+        term tuple = term_get_list_head(tuple_list);
+
+        if (!term_is_tuple(tuple)) {
+            tuple_list = term_get_list_tail(tuple_list);
+            continue;
+        }
+
+        int tuple_size = term_get_tuple_arity(tuple);
+
+        if (n_pos > tuple_size) {
+            tuple_list = term_get_list_tail(tuple_list);
+            continue;
+        }
+
+        term nth_element = term_get_tuple_element(tuple, n_pos - 1);
+
+        TermCompareResult cmp_result = term_compare(nth_element, key, TermCompareExact, ctx->global);
+
+        if (cmp_result == TermEquals) {
+            return tuple;
+        }
+        if (UNLIKELY(cmp_result == TermCompareMemoryAllocFail)) {
+            RAISE_ERROR(OUT_OF_MEMORY_ATOM);
+        }
+
+        tuple_list = term_get_list_tail(tuple_list);
+    }
+
+    return FALSE_ATOM;
+}
+
+static term nif_lists_keysearch(Context *ctx, int argc, term argv[])
```

Comment:
This function is deprecated, I'm not sure it is worth it providing a C implementation.

Reply:
Caller do not care if a function is provided using a NIF. Whether a function is implemented as a NIF or not is totally transparent to the caller.
For this specific reason it is not an argument if the BEAM has a certain function implemented as a NIF or not.

Reasons for writing a NIF are:
- performances
- access to native/operating systeam features

So, TL;DR I don't think performance really matters for deprecated functions (except specific situations), so it can be implemented in Erlang, and honestly less C code we have, more I feel happy.